packages/common/src/types/transactionEvent.ts

```diff
@@ -0,0 +1,312 @@
+    | MemoEvent
+    | TransferredWithScheduleEvent
+    | AccountCreatedEvent
+    | AmountAddedByDecryptionEvent
+    | EncryptedSelfAmountAddedEvent
+    | NewEncryptedAmountEvent
+    | EncryptedAmountsRemovedEvent
+    | DataRegisteredEvent
+    | ContractInitializedEvent
+    | ModuleDeployedEvent
+    | CredentialDeployedEvent
+    | CredentialKeysUpdatedEvent
+    | CredentialsUpdatedEvent
+    | DelegatorEvent
+    | DelegationSetDelegationTargetEvent
+    | DelegationSetRestakeEarningsEvent
+    | DelegationStakeChangedEvent
+    | BakerAddedEvent
+    | BakerRemovedEvent
+    | BakerStakeChangedEvent
+    | BakerSetRestakeEarningsEvent
+    | BakerKeysUpdatedEvent
+    | BakerSetOpenStatusEvent
+    | BakerSetMetadataURLEvent
+    | BakerSetFinalizationRewardCommissionEvent
+    | BakerSetBakingRewardCommissionEvent
+    | BakerSetTransactionFeeCommissionEvent
+    | UpdateEnqueuedEvent;
+
+// Contract Events
+
+export interface InterruptedEvent {
+    tag: TransactionEventTag.Interrupted;
+    address: ContractAddress;
+    events: string[];
+}
+
+export interface ResumedEvent {
+    tag: TransactionEventTag.Resumed;
+    address: ContractAddress;
+    success: boolean;
+}
+
+export interface UpdatedEvent {
+    tag: TransactionEventTag.Updated;
+    address: ContractAddress;
+    instigator: AddressAccount;
+    amount: bigint;
+    message: string;
+    receiveName: string;
+    events: string[];
+}
+
+export interface DataRegisteredEvent {
+    tag: TransactionEventTag.DataRegistered;
+    data: string;
+}
+
+export interface ContractInitializedEvent {
+    tag: TransactionEventTag.ContractInitialized;
+    address: ContractAddress;
+    amount: bigint;
+    contractName: string;
+    events: string[];
+}
+
+export interface ModuleDeployedEvent {
+    tag: TransactionEventTag.ModuleDeployed;
+    contents: string;
+}
+
+// Account and transfer Events
+
+export interface TransferredEvent {
+    tag: TransactionEventTag.Transferred;
+    amount: bigint;
+    to: AddressAccount;
+    from: AddressAccount;
+}
+
+export interface TransferredWithScheduleEvent {
+    tag: TransactionEventTag.TransferredWithSchedule;
+    to: AddressAccount;
+    from: AddressAccount;
+    amount: ReleaseSchedule[];
+}
+
+export interface MemoEvent {
+    tag: TransactionEventTag.TransferMemo;
+    memo: string;
+}
+
+export interface AccountCreatedEvent {
+    tag: TransactionEventTag.AccountCreated;
+    account: string;
+}
+
+export interface AmountAddedByDecryptionEvent {
+    tag: TransactionEventTag.AmountAddedByDecryption;
+    account: string;
+    amount: bigint;
+}
+
+export interface EncryptedSelfAmountAddedEvent {
+    tag: TransactionEventTag.EncryptedSelfAmountAdded;
+    account: string;
+    amount: bigint;
+    newAmount: string;
+}
+
+export interface EncryptedAmountsRemovedEvent {
+    tag: TransactionEventTag.EncryptedAmountsRemoved;
+    account: string;
+    inputAmount: string;
+    newAmount: string;
+    upToindex: number;
```

Comment:
Shouldn't it be:

```suggestion
    upToIndex: number;
```

?